### PR TITLE
Packaging & CI improvements

### LINKED ISSUE #16: Add CPU and GPU options to setup.py
Sonnet and Tensorflow have both different packages with and without GPU support. Our `setup.py` should be able to handle both cases.

Checkout [Sonnet `setup.py`](https://github.com/deepmind/sonnet/blob/master/setup.py.tmpl)

### LINKED ISSUE #50: Share builds configurations in CircleCI

### PR DESCRIPTION
**Pull request by dekked**
* Follow good practices for Python packages (see [here](https://hynek.me/articles/sharing-your-labor-of-love-pypi-quick-and-dirty)). Open for discussion!
* Use Travis CI.
* Add Code coverage.

Missing:
- [ ] easy upload to PyPI (https://github.com/tryolabs/luminoth/issues/53)
- [ ] test with GPU TensorFlow & Sonnet (does the installation really use those if they are pre-installed?)
- [ ] unit tests should run against the installed version of the package (not local repo)
- [ ] test Google Cloud integration is working (@psoto ?)

Closes #50, closes #16.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,32 +1,121 @@
...
+    'Intended Audience :: Developers',
+    'Intended Audience :: Education',
+    'Intended Audience :: Science/Research',
+    'Natural Language :: English',
+    'License :: OSI Approved :: BSD License',
+    'Topic :: Scientific/Engineering :: Mathematics',
+    'Topic :: Software Development :: Libraries :: Python Modules',
+    'Topic :: Software Development :: Libraries',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 2.6',
```
Comment (vierja) [inline setup.py]: ...

File: README.md [hunk 2]
```diff
@@ -30,28 +30,40 @@ It is indisputable that TensorFlow is currently the most mature Deep Learning fr
 [Sonnet](https://deepmind.github.io/sonnet/) fits perfectly with our mission to build code that is easy to follow and to extend. It is tricky to build a computation graph that is abstract and low-level at the same time to allows us to build complex models, and luckily Sonnet is a library that provides just that.
 
 # Installation
+Luminoth currently supports Python 2.7 and 3.4–3.6.
 
-Luminoth currently supports for both Python3 and Python2.7.
+If [TensorFlow](https://www.tensorflow.org) and [Sonnet](https://github.com/deepmind/sonnet) are already installed, Luminoth will use those versions.
 
-To install clone the repo on your machine:
+You can check the installation is working by running `lumi --help`.
```
Comment (vierja) [inline README.md]: Move to after install, and change to "worked"?

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[bdist_wheel]
+universal = 1
+
+[check-manifest]
+ignore =
+    .circleci
```
Comment (vierja) [inline setup.cfg]: Remove

File: setup.py [hunk 4]
```diff
@@ -1,32 +1,121 @@
...
+
 from setuptools import find_packages, setup
 
+
+# -------------------------------------------------------------
+
+NAME = 'luminoth'
+PACKAGES = find_packages(where='src')
+META_PATH = os.path.join('src', 'luminoth', '__init__.py')
+KEYWORDS = ['tensorflow', 'computer vision', 'object detection', 'rcnn', 'faster rcnn']
```
Comment (vierja) [inline setup.py]: remove RCNN, add 'toolkit' and 'deep learning'

File: src/luminoth/__init__.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+__version__ = '0.0.1.dev0'
+
+__title__ = 'Luminoth'
+__description__ = 'Computer vision toolkit based on TensorFlow'
+__uri__ = 'http://luminoth.ai'
+__doc__ = __description__ + ' <' + __uri__ + '>'
+
+__author__ = 'Tryolabs'
+__email__ = 'hello@tryolabs.com'
```
Comment (vierja) [inline src/luminoth/__init__.py]: luminoth@tryolabs.com 🙄

File: setup.py (hunk 1 above)
Comment (dekked) [inline setup.py]: Oops :P

**Comment by dekked** [comment]
TODO: Fix Google Cloud

@psoto this was hardcoded `gs://luminoth-config/luminoth-0.0.1-py2-none-any.whl` in `gcloud.py`. Now can use `__version__` to get the name of the wheel :)

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+- '3.4'
+- '3.5'
+- '3.6'
+install: pip install tox-travis
+script: tox
+notifications:
+  slack:
+    rooms:
+      secure: piy/NEf36gPZqw4nTAhs6dZ7Af2ozOt47RdnTSaed3tkwMfXJ+M3nccuCP4UfwmBP7fRRq2yfhLWCKIZagsFGAxhNZ/mSYgRdXfNUUofOWIJdR4+tQuqhxwgQkTfJjAWlYA+gA+GxrOnSFQ3ACRnfd/dUfthokqq7CWOzQtYYHGBSA1rAxuJU1x2qWbl+1tXa7BUJXCsR8ONz7vWDna4Znk+wixG3enOdd2pdrP2dqp9g0eUztySIT2zdjA2CB4iFIIqQLJ5fz3ab8Vpk4H9+JECB7lE8+SJVS9ZbBu8vuD9yZARBJGCPyD8HcwHTNrvlTu/V9p9W+6OeNCUXamZFuCkMHIwLgDxUntG/IyCptJqtG1n/7dOxikY1kfAIIcwMsmTQDej3xXHny1wyGMztD84w23KhNEPMJP7HsLVHmt26TIiT/yFeIOzGnh2ZPenMwtn7ti8uiweEpAH2G3tmH6bShv8myXEp++RcblSNeowHpyFhRsoevuqpT+dd+C+as2Y1DWBSFqvrfCro7AKloErJHtS3GXUiAf1Fi+Vys/sj+PLQFBh88GR5OY0lDC1cTfA4iy3Vy0IICRIylkQ4W0jfbgKF6B40syA6rM0mlu1v15P9oJvI+P5N9/G8jhUA54Ku3Xxmd6qpzlFvx2ivxmPBfHNc7g7MHJZFlzlzvk=
+    on_success: change
```
Comment (vierja) [inline .travis.yml]: Want you configure it only to "Failed" or "Recovered"?

Comment (dekked) [inline .travis.yml]: I am not sure what you mean. See [the docs](https://docs.travis-ci.com/user/notifications/#Configuring-slack-notifications).

Right now, will notify every time a build fails, and the first time it recovers. If more builds pass, then it won't notify again.

File: setup.py [hunk 7]
```diff
@@ -1,32 +1,123 @@
...
+
+#
+# If TensorFlow or Sonnet are not installed, we might as well do that.
+# Use the CPU versions by default. If the user wants to use the versions
+# with GPU support, they must be installed in advance.
+#
+try:
+    import tensorflow
+except ImportError:
+    INSTALL_REQUIRES += ['tensorflow>=1.2.1']
```
Comment (nagitsu) [inline setup.py]: Should be `tensorflow>=1.3`.

File: setup.py [hunk 8]
```diff
@@ -1,32 +1,123 @@
...
-        'dm-sonnet==1.10',
-        'google-api-python-client==1.6.2',
-        'google-cloud-storage==1.2.0',
-        'Flask==0.12.2',
-    ],
+    install_requires=INSTALL_REQUIRES,
+    test_requires=TEST_REQUIRES,
+    extras_require={
+        'gpu support': [
+            'tensorflow-gpu>=1.2.1',
```
Comment (nagitsu) [inline setup.py]: `tensorflow-gpu>=1.3`

**Pull request merged**